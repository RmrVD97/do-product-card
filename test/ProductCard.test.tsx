import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard } from '../src/components/ProductCard';
import { product1 } from './products';


const { act } = renderer;

describe('ProductCard', () => {
    
    test('debe de mostrar el componente correctamente ', () => {

        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {() => (
                        <h1>Hola </h1>
                )}
            </ProductCard>
        )
        
        expect( wrapper.toJSON() ).toMatchSnapshot();
        
    }) ;

    test('debe de mostrar el componente correctamente ', () => {

        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                ({ increaseBy,count }) => (
                      <>  <h1>Hola </h1>
                        <span>{count}</span>
                        <button onClick={() =>increaseBy(1)}></button>
                </>)
                }
            </ProductCard>
        )

        let tree = wrapper.toJSON();        
                
        expect(tree).toMatchSnapshot();
        console.log((tree as any).children[2]);
                act(() => {
                    (tree as any).children[3].props.onClick();
                })

        
        tree = wrapper.toJSON();
                
        expect((tree as any).children[2].children[0]).toBe('1');

    }) ;

    



}

)