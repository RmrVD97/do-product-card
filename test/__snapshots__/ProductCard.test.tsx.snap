// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard debe de mostrar el componente correctamente  1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Hola 
  </h1>
</div>
`;

exports[`ProductCard debe de mostrar el componente correctamente  2`] = `
<div
  className="productCard undefined"
>
    
  <h1>
    Hola 
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
